### ISSUE TITLE
ColorPalette: Clear button doesn't show when default colors array is empty

### ISSUE THREAD

**Issue by xconsau** [WordPress/gutenberg#45928]
### Description

I am developing a custom Gutenberg block, and using the [ColorPalette](https://developer.wordpress.org/block-editor/reference-guides/components/color-palette/) component for some color attributes. The ColorPallete doesn't show the clear button if the `colors` props is empty or not set. The `clearable` property is not honored in this case. Whereas if the `colors` is set to at least one default color object, the clearable also works as expected.

### Step-by-step reproduction instructions

In order to reproduce the issue, you need to add the following ColorPalette code in any test Gutenberg Block's edit.js file:

```jsx
<ColorPalette
    value={ color }
    onChange={ ( color ) => setColor( color ) }
    clearable = {true} // Regardless of true or false, this prop is not honored
/>
```

The following code will work when the colors prop is set to at least one default color object:

```jsx
<ColorPalette
    colors={ [
        { name: 'Red', color: '#f00' }
    ]}
    value={ color }
    onChange={ ( color ) => setColor( color ) }
/>
```

### Screenshots, screen recording, code snippet

![colorpalette](https://user-images.githubusercontent.com/11784237/202971872-d20e4fdf-4c4c-400e-91cc-87fcaafed503.gif)

### Environment info

- WordPress v6.1.1. Gutenberg as comes with this version of WordPress
- Chrome v 107.0.5304.107
- Desktop with Windows 10

Similar issue was addressed in #40960 but in this case the `clearable` props is not honored. So that one doesn't solve the issue.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by Mamaduka** [comment]
@ciampo, @mirka, do you know if this is a regression with the `ColorPalette` component?

**Comment by mirka** [comment]
Yes, looks like this regressed in the recent TypeScript migration. We should add a test for this.

https://github.com/WordPress/gutenberg/blob/d7e3cc12afac33be0b5b6eb9c85a27506cab4e5d/packages/components/src/color-palette/index.tsx#L87-L89

Thanks for reporting, @xconsau!

**Comment by ciampo** [comment]
Fix in #46001
